plugins/snapshots/overlay/overlay.go

```diff
@@ -655,3 +699,36 @@ func supportsIndex() bool {
 	}
 	return false
 }
+
+func findQuotaSetter(root string) (QuotaSetter, error) {
+	var backingFs = "<unknown>"
+	fsMagic, err := fs.GetMagic(root)
+	if err != nil {
+		log.G(context.TODO()).WithError(err).Warnf("get path %v filesystem magic number failed", root)
+		return nil, err
+	}
+
+	//TODO: check kernel version, kernel should be >= v4.5
+	if fsName, ok := fs.FsNames[fsMagic]; ok {
+		backingFs = fsName
+	}
+
+	switch fsMagic {
+	case fs.MagicXfs:
+		quotaCtl, err := quota.NewControl(root)
+		if err != nil {
+			log.G(context.TODO()).WithError(err).Warnf("could not initinal quota control")
+			return nil, err
+		}
+		return func(targets []string, size uint64) error {
+			if size == 0 {
+				return nil
+			}
+			return quotaCtl.SetAllQuota(quota.Size(size), targets...)
+		}, nil
+	default:
+		log.G(context.TODO()).WithError(err).Warnf("filesystem %s does not support set quota", backingFs)
+		return nil, fmt.Errorf("filesystem %s does not support set quota", backingFs)
```

Comment:
For file systems that do not support the quota mount option (ext4, etc.), do not return err here?